src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Collections.Generic;
+
+namespace Iot.Device.Tlc1543
+{
+    /// <summary>
+    /// Add documentation here
+    /// </summary>
+    public class Tlc1543 : IDisposable
+    {
+        #region Members
+
+        /// <summary>
+        /// Channel used between readings to charge up capacitors in ADC
+        /// </summary>
+        public Channel ChargeChannel = Channel.SelfTest512;

```

Comment:
please use { get; set; } for public properties like this one. Also, seems like this is more a readonly if I understand correctly or that if you change it, it you may not have all the channels possible, right? In this case, please adjust this property.

Reply:
ChargeChannel is only used between reading channel (or at the end of reading list of channels) because this device gives out data on the second cycle so having dummy value being written there helps in keeping it tidy - also helps with negating interferences between fast channel polling by reseting to one of three values 0 (SelfTest0), 512 (SelfTest512) or 1023(SelfTest1023) or if someone has a reason - there is also possibility to use one of the normal channels as a charge channel

Reply:
OK, so if only one of the 3 are usable, then please check the set and raise an ArugumentException if one of the 3 values are not used.
This would then looks like:
```csharp
private Channel _chargeChannel;
public Channel ChargeChannel
{
  get => _chargeChannel;  
  set
  {
    if ((value == Channel.SelfTest512) || (value == Channel.SelfTest0) || (value == Channel.SelfTest1023))
    {
      _chargeChannel = value;
    }
    else
    {
      thro new ArgumentException($"{ChargeChannel} can only be {Channel.SelfTest512} or {Channel.SelfTest0} or {Channel.SelfTest1023}");
    }
  }
}
```

Reply:
I've made it like this cause if someone has the reason to use normal channel as a charge channel they will be able to. It isn't forbidden to use them in this way, so i see no reason to limit them to only selftest channels
```c#
private Channel _chargeChannel = Channel.SelfTest512;
public Channel ChargeChannel
        {
            get => _chargeChannel;
            set
            {
                if ((int)value >= 0 & (int)value < 14)
                {
                    _chargeChannel = value;
                }
                else
                {
                    throw new ArgumentOutOfRangeException("ChargeChannel", "Out of range. \nMust be non-negative and less than 14. \nRefer to table 2 and 3 in TLC1543 documentation for more information");
                }
            }
        }
```